addons/web/static/tests/webclient/actions/target_tests.js

```diff
@@ -457,6 +457,47 @@ QUnit.module("ActionManager", (hooks) => {
         assert.containsOnce(webClient, ".o_kanban_view");
     });
 
+    QUnit.test(
+        'dialog will only open once for two rapid actions with the target="new"',
+        async function (assert) {
+            serverData.actions[333] = {
+                id: 333,
+                name: "A window action",
+                res_model: "partner",
+                target: "new",
+                type: "ir.actions.act_window",
+                views: [[333, "form"]],
+            };
+            serverData.actions[444] = {
+                id: 444,
+                name: "A other window action",
+                res_model: "partner",
+                target: "new",
+                type: "ir.actions.act_window",
+                views: [[444, "form"]],
+            };
+            serverData.views["partner,333,form"] = `
+            <form>
+                first action dialog
+            </form>`;
+            serverData.views["partner,444,form"] = `
+            <form>
+                second action dialog
+            </form>`;
+
+            const webClient = await createWebClient({ serverData });
+
+            doAction(webClient, 333);
+            await new Promise(resolve => setTimeout(resolve));
```

Comment:
This isn't robust enough: there's no guarantee that the test encodes the faulty behavior (i.e. the first dialog might already be there when you do the second doAction).

You could easily turn it more robust by slowing down the "onchange" rpc:
```js
const def = makeDeferred();

// mockRpc is an argument of createWebClient
mockRpc(route, args) {
    if (args.method === "onchange") {
       return def;
    }
}

doAction(webClient, 333);
await nextTick();
assert.containsNone(".o_dialog");

doAction(webClient,444);
await nextTick();
assert.containsNone(".o_dialog");

def.resolve();
await nextTick();
assert.containsOnce(".o_dialog");
```

Reply:
Also, I would move this test in actions/concurrency_tests.js

Reply:
Hello :wave: @aab-odoo Changes hvae been done.
Thank you.